presto-main/src/main/java/com/facebook/presto/metadata/SignatureBinder.java

```diff
@@ -20,6 +20,7 @@
 import com.facebook.presto.spi.type.TypeSignature;
 import com.facebook.presto.spi.type.TypeSignatureParameter;
 import com.facebook.presto.sql.analyzer.TypeSignatureProvider;
+import com.facebook.presto.type.FunctionType;
```

Comment:
Add tests to `TestSignatureBinder`. I would test things like:
`f(function<int, int>)` binds to `f(x -> cast(x as smallint))`
`f(function<int, int>)` does not binds to `f(x -> cast(x as bigint))`
`f(function<int, int>...)` to make sure varargs works
`f(function<T, U>, function<U, V>, T):V` binds to `f(x -> x + 1.0, y -> cast(y as varchar), 0)`
Also check things with parametric types, like `f(function<T, array<T>>, T)`